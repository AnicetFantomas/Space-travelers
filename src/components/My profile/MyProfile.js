import React from 'react';
import { useSelector } from 'react-redux';
import './myProfile.css';

const MyProfile = () => {
  const getAllmissions = useSelector((state) => state.missions);
  const getAlljoins = getAllmissions.filter((mission) => mission.joined);

  return (
    <div className="parent-container">
      <div className="myProfile-container">
        <h2>My missions</h2>
        <ul className="joined-missions">
          {getAlljoins.length >= 1 ? getAlljoins.map((mission) => (
            <li key={mission.name} className="mission-el">
              {mission.name}
              {' '}
              {mission.manufacturer}
            </li>
          )) : <li className="mission-el">You are not part of any mission</li> }
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
