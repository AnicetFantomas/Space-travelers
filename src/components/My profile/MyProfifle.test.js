import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Navigation/configuration';
import MyProfile from './MyProfile';

describe('test missions component', () => {
    it('Render missions row', () => {
        const app = renderer
        .create(
            <Provider store={store}>
                <MyProfile />
            </Provider>
        ).toJSON();
        expect(app).toMatchSnapshot()
    })
})