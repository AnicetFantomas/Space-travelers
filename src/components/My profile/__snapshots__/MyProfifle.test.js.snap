// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test missions component Render missions row 1`] = `
<div>
  <div
    className="myProfile-container"
  >
    <h2>
      My missions
    </h2>
    <ul
      className="joined-missions"
    >
      <li
        className="mission-el"
      >
        You are not part of any mission
      </li>
    </ul>
  </div>
</div>
`;
